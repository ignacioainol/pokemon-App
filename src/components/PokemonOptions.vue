<template>
  <h2>Pokemon Options</h2>
</template>

<script>
export default {

}
</script>

<style>

</style>