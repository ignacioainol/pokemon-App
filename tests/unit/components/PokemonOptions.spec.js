import PokemonOptions from "@/components/PokemonOptions";
import { shallowMount } from "@vue/test-utils";
import { pokemons } from "../mocks/pokemons.mock";

describe("Pokemon Options Components", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
      },
    });
  });

  test("debe de hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('debe de mostrar las 4 opciones correctamente', () => {

    const liTags = wrapper.findAll('li')
    expect(liTags.length).toBe(4)
    expect(liTags[0].text()).toBe('Bulbasaur')

  })

  test('debe de emitir "selection" con sus respectivos parametros al hacer click', () => {
    const [li1, li2, li3, li4] = wrapper.findAll('li')

    li1.trigger('click')

    console.log(wrapper.emitted('selectionPokemon'))

    // expect(wrapper.emitted('selectionPokemon'))
  })
});
