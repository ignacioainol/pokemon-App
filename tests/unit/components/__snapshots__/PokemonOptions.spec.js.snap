// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Options Components debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>Bulbasaur</li>
    <li>Ivysaur</li>
    <li>Venusaur</li>
    <li>Charmander</li>
  </ul>
</div>
`;
